crates/ui/src/windows/event_edit.rs

```diff
@@ -22,55 +22,256 @@
+                                    page.move_type == luminol_data::rpg::MoveType::Custom,
+                                    egui::Button::new("Move Route..."),
+                                ); // TODO
+                                ui.horizontal(|ui| {
+                                    ui.label("Move Speed");
+                                    luminol_components::EnumComboBox::new(
+                                        id_source.with("move_speed"),
+                                        &mut page.move_speed,
+                                    )
+                                    .ui(ui);
+                                });
+                                ui.horizontal(|ui| {
+                                    ui.label("Move Frequency");
+                                    luminol_components::EnumComboBox::new(
+                                        id_source.with("move_frequency"),
+                                        &mut page.move_frequency,
+                                    )
+                                    .ui(ui);
+                                });
+                                ui.add_space(ui.available_height());
+                            });
+                        });
+                    });
+
+                    ui.columns(2, |columns| {
+                        let [left, right] = columns else {
+                            unreachable!()
+                        };
+
+                        left.label("Options");
+                        left.group(|ui| {
+                            ui.style_mut().wrap = Some(false);
+                            ui.checkbox(&mut page.walk_anime, "Move Animation");
+                            ui.checkbox(&mut page.step_anime, "Stop Animation");
+                            ui.checkbox(&mut page.direction_fix, "Direction Fix");
+                            ui.checkbox(&mut page.through, "Through");
+                            ui.checkbox(&mut page.always_on_top, "Always on Top");
+                        });
+
+                        right.label("Trigger");
+                        right.group(|ui| {
+                            luminol_components::EnumRadioList::new(&mut page.trigger).ui(ui);
+                        });
+                    });
+                });
+
+                egui::TopBottomPanel::bottom(id_source.with("bottom_panel")).show_inside(
+                    ui,
+                    |ui| {
+                        ui.add_space(1.0); // it still looks a little squished, even with this. how can we fix this?
+                        luminol_components::close_options_ui(ui, open, &mut needs_save)
+                    },
+                );
+            });
+
+        if needs_save {
+            self.event.extra_data.modified.set(true);
+            let mut map = update_state.data.get_map(self.map_id);
+            map.events.insert(self.event.id, self.event.clone()); // don't like the extra clone, but it's necessary
+        }
```

Comment:
When saving a modified event, the new event that gets inserted into `map.events` needs to have its position updated to account for any changes to the position to the event made in between when the event editor window was opened and when the event was saved.

```suggestion
        if needs_save {
            self.event.extra_data.modified.set(true);
            let mut map = update_state.data.get_map(self.map_id);
            if let Some(old_event) = map.events.get(self.event.id) {
                self.event.x = old_event.x;
                self.event.y = old_event.y;
            }
            map.events.insert(self.event.id, self.event.clone()); // don't like the extra clone, but it's necessary
        }
```

Although maybe it would be better for the event editor to just get a mutable reference to the event it's editing from `update_state` every frame.